evm/src/cpu/kernel/asm/curve/bn254/field_arithmetic/power.asm

```diff
@@ -0,0 +1,229 @@
+global test_pow:
+    // stack: ptr, f, ptr, out, return_fp12_on_stack, out
+    %store_fp12
+    // stack:         ptr, out, return_fp12_on_stack, out
+    %jump(power)
+
+/// def power(acc):
+///     power_init()
+///     power_loop_4()
+///     power_loop_2()
+///     power_loop_0()
+///     power_return()
+///
+/// def power_init()
+///     y0, y4, y2 = 1, 1, 1
+///
+/// def power_return()
+///     y0  = y0^{-1}
+///     y4 *= y0 * (y2**2)
+///     y4  = frob_fp12_1(y4)
+///     y2  = frob_fp12_2_(y2)
+///     return y2 * y4 * y0 
+
+global power:
```

Comment:
Thank you for including the Python above. But please explain what this function does in English. `power` could be anything.

Also, please include a bit more details (i.e. types) in the name if you're going to export it as a global. There's many different things that could be called `power`.